paasta_tools/frameworks/native_scheduler.py

```diff
@@ -78,7 +88,7 @@ def __init__(self, service_name, instance_name, cluster,
                  system_paasta_config, soa_dir=DEFAULT_SOA_DIR,
                  service_config=None, reconcile_backoff=30,
                  instance_type='paasta_native', service_config_overrides={},
-                 reconcile_start_time=float('inf')):
+                 reconcile_start_time=float('inf'), constraints=[]):
```

Comment:
mutable default args :( 

```
>>> class X:
...    def __init__(self, foo=[]):
...        self.foo = foo
...        print foo
... 
>>> x1 = X()
[]
>>> x1.foo.append('y')
>>> x1.foo
['y']
>>> x2 = X()
['y']
>>> x2.foo.append('foo')
>>> x2.foo
['y', 'foo']
```

you'll have to set the default arg as ``None`` and do ``if not constraints: self.constraints = []``